sale_order_deliverable_rate/tests/test_sale_order.py

```diff
@@ -0,0 +1,156 @@
+
+from odoo.tests import Form
+from odoo.tests.common import SavepointCase
+
+
+class TestSaleOrderDeliverableRate(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        cls.WizardQty = cls.env["stock.change.product.qty"]
+
+        cls.Partner = cls.env["res.partner"]
+        cls.Product = cls.env["product.product"]
+        cls.SaleOrder = cls.env["sale.order"]
+        cls.SaleOrderLine = cls.env["sale.order.line"]
+
+        cls.partner = cls.Partner.create({"name": "Customer"})
+        cls.array_products = [
+            {
+                "product": cls.Product.create(
+                    {"name": "Test Product 1", "type": "product"}
+                ),
+                "product_uom_qty": 10,
+                "qty_delivered": 0,
+            },
+            {
+                "product": cls.Product.create(
+                    {"name": "Test Product 2", "type": "product"}
+                ),
+                "product_uom_qty": 10,
+                "qty_delivered": 0,
+            },
+            {
+                "product": cls.Product.create(
+                    {"name": "Test Product 3", "type": "product"}
+                ),
+                "product_uom_qty": 10,
+                "qty_delivered": 0,
+            },
+        ]
+
+    @classmethod
+    def _update_stock(cls, product, qty):
+        """
+        Set the stock quantity of the product
+        :param product: product.product recordset
+        :param qty: float
+        """
+        wizard = Form(cls.WizardQty)
+        wizard.product_id = product
+        wizard.product_tmpl_id = product.product_tmpl_id
+        wizard.new_quantity = qty
+        wizard = wizard.save()
+        wizard.change_product_qty()
+
+    def create_basic_so(self, array_products):
+        return self.SaleOrder.create(
+            {
+                "partner_id": self.partner.id,
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": product["product"].id,
+                            "name": "line 1",
+                            "product_uom_qty": product["product_uom_qty"],
+                            "qty_delivered": product["qty_delivered"],
+                            "price_unit": 10,
+                        },
+                    )
+                    for product in array_products
+                ],
+            }
+        )
+
+    def test_totally_deliverable(self):
+        """
+        Ensure that deliverable_rate of an order is to 100%
+        when it contains one order_line with a product who has enough stock
+        :return:
+        """
+        self._update_stock(self.array_products[0]["product"], 10)
+        self.array_products[0]["product_uom_qty"] = 10
+        so = self.create_basic_so(self.array_products[:1])
```

Comment:
You're creating a sale order into every test (except one). So best practice is to create it once during `setUpClass()`.
But you can let it into a dedicated function.